src/Classes/ServiceAPI/MyRadio_Track.php

```diff
@@ -438,16 +438,73 @@ private static function findByNameArtist($title, $artist, $limit, $digitised = f
     }
 
     /**
+     * Search for tracks in the library.
+     *
+     * @param string $title Only return tracks matching this title
+     * @param string $artist Only return tracks matching this artist
+     * @param integer $recordid Only return tracks in this album
+     * @param boolean $digitised Only return tracks that are digitised. If false, return any. Default true.
+     * @param enum $clean Only return tracks with the given cleanliness (y = clean, n = explicit, u = unknown)
+     * @param boolean $precise Only return exact matches for title and artist. Defaults to fuzzy search.
+     * @param integer $page Search only returns 50 results by default. Increment this counter for additional results.
+     * @param enum $sort Sort order. Possible values: "id" (default), "title", "random". Random will not paginate well.
+     */
+    public static function search(
+        $title = null,
+        $artist = null,
+        $recordid = null,
+        $digitised = true,
+        $clean = null,
+        $precise = false,
+        $page = 1,
+        $sort = null
+    ) {
```

Comment:
split if^H^Hfunction declaration. this one will be a bit long, but I don't think it'll be too awful. No worse than MyRadio_User::create anyway